client/apiversion.go

```diff
@@ -67,6 +66,14 @@ func getAPIVersionURL(apiURLVersionInfo *apiURLVersion, requested apiVersion) (s
 		return "", fmt.Errorf("could not find matching URL")
 	}
 	versionURL := apiURLVersionInfo.rootURL
+
+	// https://bugs.launchpad.net/juju/+bug/1756135:
+	// some hrefURL.Path contain more than the version, with
+	// overlap on the apiURLVersionInfo.rootURL
+	if strings.HasPrefix(match, "/"+versionURL.Path) {
+		versionURL.Path = "/"
```

Comment:
Maybe I'm being verbose, but trace logging for this?

Reply:
added.